TicTacToe.c

```diff
@@ -0,0 +1,128 @@
+         case 2: Count[DIA2][player] += 1; break;  // Diagonal 2, index positions 2-4-6
+         case 4: Count[DIA1][player] += 1; Count[DIA2][player] += 1; break;
+         case 6: Count[DIA2][player] += 1; break;
+         case 8: Count[DIA1][player] += 1; break;
+      }
+   }
+   if (Count[rowNo][player] >= 3 || Count[colNo][player] >= 3 ||
+       Count[DIA1][player] >= 3 || Count[DIA2][player] >= 3) return isPlayer1 ? 1 : 2;
+   return 0;
+}
+
+// Returns first non-space input character obtained from the user.
+static char Get_Input (wchar_t* prompt) {
+   char inChar[20];
+   do {
+      wprintf (L"%s ", prompt);
+      fgets (inChar, sizeof (inChar), stdin);
+      // If the input exceeds 20, clear input buffer.
+      if (inChar[strlen (inChar) - 1] != '\n') while (getchar () != '\n');
+   } while (inChar[0] == '\n' || strlen (inChar) > 2);
+   return inChar[0];
+}
+
+// Gets two non-space unique symbols to represent player 1 and 2.
+static void Get_Symbol (char* s1, char* s2) {
+   *s1 = Get_Input (L"Enter symbol 1:");
+   *s2 = Get_Input (L"Enter symbol 2:");
+   while (*s1 == *s2) *s2 = Get_Input (L"Please enter a unique symbol. Enter symbol 2:");
+}
+
+// Prints 3 x 3 game board.
+static void PrintBoard (wchar_t* grid) {
+   wprintf (L"\u250F\u2501\u2501\u2501\u2533\u2501\u2501\u2501\u2533\u2501\u2501\u2501\u2513\n\u2503");
+   for (int i = 0, l = 0; i < 3; i++) {
+      for (int k = 0; k < 3; l++, k++) wprintf (L" %wc \u2503", grid[l]);
+      if (l >= 9) break;
+      wprintf (L"\n\u2523\u2501\u2501\u2501\u254B\u2501\u2501\u2501\u254B\u2501\u2501\u2501\u252B\n\u2503");
+   }
+   wprintf (L"\n\u2517\u2501\u2501\u2501\u253B\u2501\u2501\u2501\u253B\u2501\u2501\u2501\u251B\n");
+}
+
+// Returns the index of the box number within the range of 0-8.
+static int Get_Box_Num (wchar_t* arr, bool isAutoPlay) {
+   if (isAutoPlay) {
+      int num = 4;
+      while (arr[num] != L' ') num = rand () % 9;
+      return num;
+   }
+   while (true) {
+      char boxNum = Get_Input (L"Enter box number (1-9):");
+      if (boxNum < L'1' || boxNum > L'9' || arr[boxNum - L'1'] != L' ') {
+         wprintf (L"Entered box number is either occupied or invalid\nExiting the program.....");
+         return -1;
+      }
+      return boxNum - '1';
+   }
+}
+
+static bool ModeChoice () {
+   char choice = ' ';
+   do
+      choice = Get_Input (L"Enter (1/2):");
+   while (!(choice == '1' || choice == '2'));
+   return choice == '1';
+}
+
+int main () {
+   int val = _setmode (_fileno (stdout), _O_U8TEXT);
+   wchar_t grid[9] = { L' ',L' ',L' ' ,L' ' ,L' ' ,L' ' ,L' ' ,L' ',L' ' };
+   char P1, P2;
+   Get_Symbol (&P1, &P2);
+   bool isPlayer1 = true, isAutoPlay = false;
+   wprintf (L"Enter 1 to play against the computer. Enter 2 for dual player mode.\n");
+   if (ModeChoice ()) isAutoPlay = true;
+   for (int i = 0; i < 9; i++, isPlayer1 = !isPlayer1) {
+      wprintf (L"%ls :\n", isPlayer1 ? L"Player 1" : L"Player 2");

```

Comment:
```suggestion
      wprintf (L"Player %d :\n", isPlayer1 );

```
isPlayer1 can be an int with 1 and 2 as possible values.
Check if this simplifies implementation.